### PR TITLE
 Add action to create /_alerting/destinations/

### PR DESCRIPTION
**Pull request by skkosuri-amzn**
*Issue #, if available:*
#232

*Description of changes:*
#232
Failures responses are also standardized. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+                                override fun onResponse(response: AcknowledgedResponse) {
+                                    onUpdateMappingsResponse(response)
+                                }
+                                override fun onFailure(t: Exception) {
+                                    actionListener.onFailure(t)
+                                }
+                            })
+                } else {
+                    prepareDestinationIndexing()
+                }
```
Comment (lezzago) [inline alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt:81]: Can you change the if block to `else if`, so we wont have any nested if else statements?

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/IndexDestinationResponse.kt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import org.elasticsearch.rest.RestStatus
+import java.io.IOException
+
+class IndexDestinationResponse : ActionResponse, ToXContentObject {
+    var id: String
+    var version: Long
+    var seqNo: Long
+    var primaryTerm: Long
+    var status: RestStatus
+    var destination: Destination
```
[the remark this answers is not in the document]
Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/IndexDestinationResponse.kt:38]: Made val

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/IndexDestinationRequest.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import org.elasticsearch.rest.RestRequest
+import java.io.IOException
+
+class IndexDestinationRequest : ActionRequest {
+    var destinationId: String
+    var seqNo: Long
+    var primaryTerm: Long
+    var refreshPolicy: WriteRequest.RefreshPolicy
+    var method: RestRequest.Method
+    var destination: Destination
```
[the remark this answers is not in the document]
Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/IndexDestinationRequest.kt:33]: Made val

File: alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt (hunk 1 above)
Comment (skkosuri-amzn) [reply alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexDestinationAction.kt:81]: Good catch, changed.

**Pull request merged**